### PR TITLE
Updating our dependencies.

### PR DESCRIPTION
**Pull request by tannergooding**
This resolves https://github.com/dotnet/project-system/issues/2307.

This additionally ensures we are referencing the latest versions of our dependencies (or the current version used by VS) to ensure we are consuming the latest APIs, bug fixes, etc.

### CODE REVIEW

**Comment by tannergooding** [comment]
FYI. @dotnet/project-system

File: src/VisualStudio.props [hunk 1]
```diff
@@ -74,6 +81,19 @@
            or '%(FileName)' == 'VSLangProj110'">
         <EmbedInteropTypes>true</EmbedInteropTypes>
       </ReferencePath>
+      <ReferencePath Condition="
```
Comment (jmarolf) [inline src/VisualStudio.props]: Add comment explaining why this is necessary

Comment (tannergooding) [reply src/VisualStudio.props]: Fixed.

**Comment by tannergooding** [comment]
All issues should be resolved now.

**Comment by tannergooding** [comment]
FYI. @Pilchie for ask mode approval

**Comment by Pilchie** [comment]
Why is this necessary for ask mode?

**Comment by tannergooding** [comment]
@Pilchie, at the very least, it helps ensure unit tests are running against bits that are closer to what we will be shipping against.

Realistically, this is just infrastructure/quality-of-life improvements. If this can't go into master right now, then it will probably just wait until we snap master to a release branch and then it can go in.

File: src/VisualStudio.props (hunk 1 above)
Comment (Pilchie) [reply src/VisualStudio.props]: Comparing binaries before and after confirms that this doesn't change the set of things that are embedded?

**Comment by tannergooding** [comment]
> Comparing binaries before and after confirms that this doesn't change the set of things that are embedded?

Correct.

**Pull request merged**